zam/aspirateur/zam_aspirateur/__main__.py

```diff
@@ -0,0 +1,195 @@
+"""
+Récupérer la liste des amendements à un texte de loi au Sénat
+"""
+import argparse
+import math
+from typing import (
+    Iterable,
+    Iterator,
+    List,
+    Optional,
+)
+
+from zam_aspirateur.amendements.fetch import (
+    fetch_and_parse_all,
+    fetch_and_parse_discussed,
+)
+from zam_aspirateur.amendements.models import Amendement
+from zam_aspirateur.amendements.writer import (
+    write_csv,
+    write_xlsx,
+)
+from zam_aspirateur.senateurs.fetch import fetch_senateurs
+from zam_aspirateur.senateurs.parse import parse_senateurs
+
+
+def main(argv: List[str] = None) -> None:
+    args = parse_args(argv=argv)
+
+    print(f'Récupération des amendements déposés...')
+    amendements = fetch_and_parse_all(
+        session=args.session,
+        num=args.texte,
+    )
+
+    processed_amendements = process_amendements(
+        amendements=amendements,
+        session=args.session,
+        num=args.texte,
+    )
+
+    save_output(
+        amendements=processed_amendements,
+        basename=f'amendements_{args.session}_{args.texte}',
+        format=args.output_format,
+    )
+
+
+def parse_args(argv: List[str] = None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--session',
+        required=True,
+        help='session parlementaire (p.ex. 2017-2018)',
+    )
+    parser.add_argument(
+        '--texte',
+        required=True,
+        help='numéro du texte au Sénat (p.ex. 330)',
+    )
+    parser.add_argument(
+        '--output-format',
+        default='csv',
+        choices=['csv', 'xlsx'],
+        help='format du tableau à générer',
+    )
+    return parser.parse_args(argv)
+
+
+def process_amendements(
+    amendements: Iterable[Amendement],
+    session: str,
+    num: str,
+) -> Iterable[Amendement]:
+
+    # Les amendements discutés en séance, par ordre de passage
+    print(f'Récupération des amendements soumis à la discussion...')
```

Comment:
Do we keep that pattern of beginning each string by `f`?

Reply:
I guess that was some bad copying and pasting. Fixed in d51291e8ae2e0b8c219da4458fc7df134c07fa1c.